### PR TITLE
Enhancement/137

### PR DESCRIPTION
**Pull request by theyashwanthsai**
### **User description**
Issue #137 , Issue #138, Issue #139

@sohamganatra If there are any changes needed to be made, i will do that. Rest of the examples can follow the same structure so shld take 30 min in finishing them

___

### **PR Type**
Enhancement, Documentation

___

### **Description**
- Enhanced the commit agent to use environment variables for configuration, improving flexibility and security.
- Added an example `.env` file to guide users in setting up necessary environment variables.
- Updated the README with comprehensive setup and usage instructions, including steps for creating a virtual environment, installing dependencies, and configuring triggers.
- Added a `requirements.txt` file listing all necessary dependencies for the project.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>agent.py</strong><dd><code>Enhance commit agent with environment variables and improved structure</code></dd></summary>
<hr>
      
examples/commit_agent/agent.py

<li>Added environment variable loading using <code>dotenv</code>.<br> <li> Replaced hardcoded values with environment variables.<br> <li> Updated agent initialization and task definitions.<br> <li> Improved comments and code formatting.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/142/files#diff-8b28e2132b94e46b8a32f9cc399aceca0b93f85c7154fd6c6b420d33e3a8c099">+35/-19</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>.env.example</strong><dd><code>Add example environment variables file for configuration</code>&nbsp; </dd></summary>
<hr>
      
examples/commit_agent/.env.example

<li>Added example environment variables for OpenAI API key and Trello list <br>IDs.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/142/files#diff-e5cc1fef7b53ef2ec1e1592582a8e5cd19d7f20f0aff4cfbedd0dd335cc4de7b">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Add README with setup and usage instructions for commit agent</code></dd></summary>
<hr>
      
examples/commit_agent/README.md

<li>Added detailed instructions for setting up and running the commit <br>agent.<br> <li> Included steps for creating a virtual environment, installing <br>dependencies, and configuring triggers.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/142/files#diff-31b73b178daa8bf56d12afaba41ec8fe0ed97f01bd45ccfe2aa79422eb108ba7">+43/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Add requirements.txt with necessary dependencies</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/commit_agent/requirements.txt

- Added required dependencies for the commit agent project.

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/142/files#diff-53cbd419a40f18c0d25289e26f1337c69c85bcd62a192a61863a9f7f502371b9">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 83f636b1c3f035addc6106702d1a4b1795727568  | 
|--------|--------|

### Summary:
Introduces the 'Commit Agent' example project to sync GitHub commits with Trello boards, including setup instructions and necessary scripts.

**Key points**:
- Added `examples/commit_agent/.env.example` for environment setup.
- Created `examples/commit_agent/README.md` with detailed setup instructions.
- Implemented `examples/commit_agent/agent.py` to sync GitHub commits with Trello.
- Defined dependencies in `examples/commit_agent/requirements.txt`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by sohamganatra** [comment]
Let's add a setup.sh file. It will do following things 

1. Create a virtual env 
2. Install all the required packages 
3. Activate the env
4. Copy env backup to env file
5. prompt user to fill the env file 

Once user does that, they can simply run the main file.

**Comment by sohamganatra** [comment]
Are we sticking to agent.py? Or we can just name it main.py. That gives me clear idea about running it.

**Comment by theyashwanthsai** [comment]
@sohamganatra 

> Let's add a setup.sh file. It will do following things
> 
> 1. Create a virtual env
> 2. Install all the required packages
> 3. Activate the env
> 4. Copy env backup to env file
> 5. prompt user to fill the env file
> 
> Once user does that, they can simply run the main file.

Finished the changes, please look into it

File: examples/competitor_researcher/setup.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+echo "Activating virtual environment..."
+source ~/.venvs/competitor_researcher/bin/activate
+
+# Install libraries from requirements.txt 
+echo "Installing libraries from requirements.txt..."
+pip install -r requirements.txt
+
+# Login to your account
+echo "Login to your Composio acount"
+composio composio login
```
Comment (sohamganatra) [inline examples/competitor_researcher/setup.sh:17]: This line doesn't look correct. Have all this scripts been tested?

File: examples/calendar_agent/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Calendar Agent
+> Fork and Clone this repository if needed!
+
+## Introduction
+This project is an example which uses Composio to seamlessly convert your to-do lists into Google Calendar events. 
+It automatically schedules tasks with specified labels and times, ensuring your calendar is always up-to-date and organized.
+
```
Comment (sohamganatra) [inline examples/calendar_agent/README.md:7]: Can we add a how it works or some explaination section. Ideally I want the person to also quickly understand how end to end code works in less then 3 steps and then how to run it. 

Overall structure -> what the code is about, how it works and how to run it.

**Review by sohamganatra — CHANGES_REQUESTED** [review]

File: examples/competitor_researcher/setup.sh (hunk 1 above)
Comment (theyashwanthsai) [reply examples/competitor_researcher/setup.sh]: I have changed, Verfied the working.

File: examples/calendar_agent/README.md (hunk 2 above)
Comment (theyashwanthsai) [reply examples/calendar_agent/README.md:7]: I think this is a good point. I will quickly add one and send a PR

Comment (theyashwanthsai) [reply examples/calendar_agent/README.md:7]: > Can we add a how it works or some explaination section. Ideally I want the person to also quickly understand how end to end code works in less then 3 steps and then how to run it.
> 
> Overall structure -> what the code is about, how it works and how to run it.

Its completed. I have added 3 steps, sort of a high level overview of how the project works.

**Review by sohamganatra — APPROVED** [review]

**Review by angrybayblade — APPROVED** [review]

**Pull request merged**